unresolved import `libc::siginfo_t`

Hello,

I wanted to try this library with my iodeo board but I got the following error when trying to compile the example app.

```
error[E0432]: unresolved import `libc::siginfo_t`
  --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:78:38
   |
78 | use libc::{c_int, c_void, sigaction, siginfo_t};
   |                                      ^^^^^^^^^ no `siginfo_t` in the root

error[E0432]: unresolved imports `libc::SIGKILL`, `libc::SIGSTOP`
  --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:83:28
   |
83 | use libc::{SIGFPE, SIGILL, SIGKILL, SIGSEGV, SIGSTOP};
   |                            ^^^^^^^           ^^^^^^^ no `SIGSTOP` in the root
   |                            |
   |                            no `SIGKILL` in the root
   |                            help: a similar name exists in the module: `SIGILL`

error[E0425]: cannot find value `SA_RESTART` in crate `libc`
   --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:172:27
    |
172 |         let flags = libc::SA_RESTART;
    |                           ^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `SA_SIGINFO` in crate `libc`
   --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:175:25
    |
175 |         siginfo = libc::SA_SIGINFO as _;
    |                         ^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `SA_SIGINFO` in crate `libc`
   --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:256:29
    |
256 |             siginfo = libc::SA_SIGINFO as _;
    |                             ^^^^^^^^^^ not found in `libc`

error[E0609]: no field `sa_sigaction` on type `sigaction`
   --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:162:15
    |
162 |         { new.sa_sigaction = handler as usize; }
    |               ^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sa_handler`, `sa_mask`, `sa_flags`

error[E0609]: no field `sa_sigaction` on type `sigaction`
   --> /home/oschijns/.cargo/registry/src/index.crates.io-6f17d22bba15001f/signal-hook-registry-1.4.2/src/lib.rs:243:30
    |
243 |         let fptr = self.info.sa_sigaction;
    |                              ^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sa_handler`, `sa_mask`, `sa_flags`

Some errors have detailed explanations: E0425, E0432, E0609.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `signal-hook-registry` (lib) due to 7 previous errors
```

It looks like it is the same error that affect the CI.

Is there a way to fix it?

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Put the invalidation group hack behind a feature flag
+
 ### Removed
 
 ## [0.3.0] - 2025-02-08

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,7 +17,6 @@ members = ["minitel-app-example"]
 
 [dependencies]
 ratatui = { version = "0.29.0", optional = true, default-features = false, features = [
-    "underline-color",
 ] }
 axum = { version = "0.8.1", default-features = false, features = [
     "ws",
@@ -37,6 +36,7 @@ ratatui = ["dep:ratatui"]
 docsrs = ["ratatui", "axum", "espdoc", "futures"]
 axum = ["dep:axum"]
 futures = ["dep:futures"]
+invalidation-group = ["ratatui/underline-color"]
 
 [dev-dependencies]
 env_logger = "0.11.6"

--- a/src/ratatui.rs
+++ b/src/ratatui.rs
@@ -306,6 +306,7 @@ pub mod border {
 
 pub trait StyledMinitelExt {
     type Item;
+    #[cfg(feature = "invalidation-group")]
     fn invalidation_group(self, group: u8) -> Self::Item;
 }
 
@@ -314,6 +315,7 @@ where
     T: Styled<Item = T>,
 {
     type Item = Self;
+    #[cfg(feature = "invalidation-group")]
     fn invalidation_group(self, group: u8) -> Self::Item {
         let style = self.style().underline_color(Color::Indexed(group));
         self.set_style(style)

--- a/src/stum/videotex.rs
+++ b/src/stum/videotex.rs
@@ -283,12 +283,12 @@ impl G1 {
     /// [5, 6]]
     pub fn from_bits(bits: [[bool; 2]; 3]) -> Self {
         let val: u8 = (bits[0][0] as u8)
-            | (bits[0][1] as u8) << 1
-            | (bits[1][0] as u8) << 2
-            | (bits[1][1] as u8) << 3
-            | (bits[2][0] as u8) << 4
-            | (true as u8) << 5
-            | (bits[2][1] as u8) << 6;
+            | ((bits[0][1] as u8) << 1)
+            | ((bits[1][0] as u8) << 2)
+            | ((bits[1][1] as u8) << 3)
+            | ((bits[2][0] as u8) << 4)
+            | ((true as u8) << 5)
+            | ((bits[2][1] as u8) << 6);
         G1(val)
     }
 

